File: Infra/Infra/PolateStop.cpp
#include "PolateStop.hpp"

CppClassNew(PolateStop)

Int PolateStop_Init(Int o)
{
    PolateStop* m;
    m = CP(o);
    Int count;
    count = m->Count;
    
    m->Intern = new QGradientStops;
    m->Intern->resize(count);
    return true;
}

Int PolateStop_Final(Int o)
{
    PolateStop* m;
    m = CP(o);
    
    delete m->Intern;
    
    return true;
}

CppField(PolateStop, Count)

Int PolateStop_PointGetPos(Int o, Int index)
{
    PolateStop* m;
    m = CP(o);

    qsizetype indexU;
    indexU = index;

    QGradientStop* uu;
    uu = (QGradientStop*)(&(m->Intern->at(indexU)));

    qreal posU;
    posU = uu->first;

    ValueFromInternValue(posU);

    Int a;
    a = posUA;
    return a;
}

Int PolateStop_PointGetColor(Int o, Int index)
{
    PolateStop* m;
    m = CP(o);

    qsizetype indexU;
    indexU = index;

    QGradientStop* uu;
    uu = (QGradientStop*)(&(m->Intern->at(indexU)));

    QColor colorU;
    colorU = uu->second;

    QRgb colorUu;
    colorUu = colorU.rgba();
    Int32 colorUa;
    colorUa = (Int32)colorUu;
    Int colorA;
    colorA = colorUa;

    Int a;
    a = colorA;
    return a;
}

Int PolateStop_PointSet(Int o, Int index, Int pos, Int color)
{
    PolateStop* m;
    m = CP(o);

    qsizetype indexU;
    indexU = index;

    InternValue(pos);

    InternColor(color);

    QGradientStop* uu;
    uu = (QGradientStop*)(&(m->Intern->at(indexU)));

    *uu = QGradientStop();

    uu->first = posU;
    uu->second = colorU;
    return true;
}

Int PolateStop_Intern(Int o)
{
    PolateStop* m;
    m = CP(o);
    Int a;
    a = CastInt(m->Intern);
    return a;
}

File: Mirai/Mirai.Draw/PolateStop.cs
namespace Mirai.Draw;

public class PolateStop : Any
{
    public override bool Init()
    {
        base.Init();
        this.DrawInfra = Infra.This;

        long count;
        count = this.Count;
        ulong countU;
        countU = (ulong)count;
        this.Intern = Extern.PolateStop_New();
        Extern.PolateStop_CountSet(this.Intern, countU);
        Extern.PolateStop_Init(this.Intern);
        return true;
    }

    public virtual bool Final()
    {
        Extern.PolateStop_Final(this.Intern);
        Extern.PolateStop_Delete(this.Intern);
        return true;
    }

    public virtual long Count { get; set; }

    protected virtual Infra DrawInfra { get; set; }
    internal virtual ulong Intern { get; set; }

    public virtual bool PointSet(long index, PolateStopPoint point)
    {
        ulong indexU;
        ulong posU;
        ulong colorU;
        indexU = (ulong)index;
        posU = (ulong)(point.Pos);
        colorU = this.DrawInfra.InternColor(point.Color);
        Extern.PolateStop_PointSet(this.Intern, indexU, posU, colorU);
        return true;
    }

    public virtual bool PointGet(long index, PolateStopPoint result)
    {
        ulong indexU;
        indexU = (ulong)index;

        ulong ua;
        ulong ub;

        ua = Extern.PolateStop_PointGetPos(this.Intern, indexU);

        ub = Extern.PolateStop_PointGetColor(this.Intern, indexU);

        long pos;
        pos = (long)(ua);
        result.Pos = pos;
        this.DrawInfra.ColorSet(result.Color, ub);
        return true;
    }
}